tests/test_tokens.py

```diff
@@ -281,6 +281,11 @@ def test_check_exp(self):
         with self.assertRaises(TokenError):
             token.check_exp('refresh_exp', current_time=current_time + timedelta(days=2))
 
+        # a token 1 day expired is valid if leeway is 2 days
+        token._token_backend.leeway = timedelta(days=2).total_seconds()
```

Comment:
I'd much rather make this a separate test where we actually input leeway instead of manipulating a private attribute.

Reply:
Yes, this definitely sounds like a good idea. I just put it there because it seemed to match the style of the tests here on first glance.

I'm not sure when I'm gonna find more time to work on this, please feel free to continue this pull request yourself if you have an idea on how to improve it.